File: src/v1/chromatic_circle/ChromaticCircle.ts
export class ChromaticCircle {
  private readonly filter: string;

  constructor(filter: string) {
    this.filter = filter;
  }

  public mixColors(color: string): string {
    if (this.filter === color) {
      return color;
    }
    if (this.filter === 'blue') {
      if (color == this.filter) {
        return color;
      }
      if (color === 'yellow') {
        return 'green';
      }
      if (color === 'red') {
        return 'violet';
      }
    }
    if (this.filter === 'red') {
      if (color === 'yellow') {
        return 'orange';
      }
      if (color === 'blue') {
        return 'violet';
      }
    }
    if (this.filter === 'yellow') {
      if (color === 'red') {
        return 'orange';
      }
      if (color === 'blue') {
        return 'green';
      }
    }

    throw new Error(`invalid color: ${color}`);
  }
}

File: flutter/lib/chromatic_circle.dart
import 'dart:ui';

class ChromaticCircle {
  final Color filter;

  ChromaticCircle({
    required this.filter,
  });

  Color mixColors(Color color) {
    if (filter == color) {
      return color;
    }

    if (filter == Colors.blue) {
      if (color == Colors.blue) {
        return color;
      }
      if (color == Colors.yellow) {
        return Colors.green;
      }
      if (color == Colors.red) {
        return Colors.purple;
      }
    }

    if (filter == Colors.red) {
      if (color == Colors.yellow) {
        return Colors.orange;
      }
      if (color == Colors.blue) {
        return Colors.purple;
      }
    }

    if (filter == Colors.yellow) {
      if (color == Colors.red) {
        return Colors.orange;
      }
      if (color == Colors.blue) {
        return Colors.green;
      }
    }

    throw ArgumentError('Invalid color: ${color.toString()}');
  }
}
